tf-ncl/src/nickel_builder.rs

```diff
@@ -156,31 +164,49 @@ impl Field<Record> {
     pub fn no_value(mut self) -> Record {
         self.record.fields.push((
             self.path,
-            Term::MetaValue(self.metadata.unwrap_or_default()).into(),
+            record::Field {
+                value: None,
+                metadata: self.metadata,
+                ..Default::default()
+            },
         ));
         self.record
     }
 
     pub fn value(mut self, value: impl Into<RichTerm>) -> Record {
-        self.record
-            .fields
-            .push((self.path, with_metadata(self.metadata, value)));
+        self.record.fields.push((
+            self.path,
+            record::Field {
+                value: Some(value.into()),
+                metadata: self.metadata,
+                ..Default::default()
+            },
+        ));
         self.record
     }
 }
 
 #[derive(Debug)]
 pub struct Record {
-    fields: Vec<(StaticPath, RichTerm)>,
+    fields: Vec<(StaticPath, record::Field)>,
     attrs: RecordAttrs,
 }
 
-fn elaborate_field_path(path: StaticPath, content: RichTerm) -> (FieldPathElem, RichTerm) {
+fn elaborate_field_path(
+    path: StaticPath,
+    content: record::Field,
+) -> (FieldPathElem, record::Field) {
     let mut it = path.into_iter();
     let fst = it.next().unwrap();
 
     let content = it.rev().fold(content, |acc, id| {
-        Term::Record(RecordData::with_fields([(id, acc)].into())).into()
+        record::Field::from(RichTerm::new(
```

Comment:
You can use `RichTerm::from(t: Term)` if you don't care about the position (will be set to `TermPos::None`).